File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        string convertInt = to_string(x);
        
        string result = "";
        
        if (convertInt[0] == '-') {
            result += convertInt[0];
        }
        
        for (int i = convertInt.size() - 1; i >= 0; i--) {
            result += convertInt[i];
        }
        
        try {
            return stoi(result);
        } catch (...) {
            return 0;
        }
    }
};

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        
        String str = String.valueOf(x);
        
        String result = "";

        Integer iterator = 0;
        if (str.charAt(0) == '-') {
            result += str.charAt(0);
            iterator = 1;
        }

        for (int i = str.length() - 1; i >= iterator; i--) {
            
            result += str.charAt(i);
        } 
        
        try {
            return Integer.parseInt(result);   
        } catch (Exception e) {
            return 0;
        }
    }
}